pkg/storage/plugin_adapter.go

```diff
@@ -167,8 +178,8 @@ func (a PluginAdapter) Remove(ctx context.Context, collection string, opts Remov
 }
 
 func (a PluginAdapter) Update(ctx context.Context, collection string, opts UpdateOptions) error {
-	pluginOpts, err := opts.ToPluginOptions(collection)
-	err = a.plugin.Update(ctx, pluginOpts)
+	pluginOpts, _ := opts.ToPluginOptions(collection)
+	err := a.plugin.Update(ctx, pluginOpts)
```

Comment:
ℹ️ `err` was not used.

Reply:
Can you have it check and return the error when not nil?